RegistryManager/src/test/java/eu/neclab/ngsildbroker/registryhandler/controller/RegistryControllerTest.java

```diff
@@ -0,0 +1,303 @@
+				+ "                    \"id\": \"urn:ngsi-ld:ContextSourceRegistration:5616438376726496\",\r\n"
+				+ "                    \"type\": \"Vehicle1\"\r\n" + "                }\r\n" + "            ],\r\n"
+				+ "            \"propertyNames\": [\r\n" + "                \"brandName\",\r\n"
+				+ "                \"speed\"\r\n" + "            ],\r\n" + "            \"relationshipNames\": [\r\n"
+				+ "                \"isParked\"\r\n" + "            ]\r\n" + "        }\r\n" + "    ] \r\n" + "}";
+
+		payload1 = "{\r\n" + "    \"id\": \" 3\",\r\n" + "    \"type\": \"Test10.3\",\r\n" + "    \"name\": {\r\n"
+				+ "        \"type\": \"Property\",\r\n" + "        \"value\": \"BMW\"\r\n" + "    },\r\n"
+				+ "    \"speed\": {\r\n" + "        \"type\": \"Property\",\r\n" + "        \"value\": 80\r\n"
+				+ "    }\r\n" + "}";
+
+		updatePayload = "{\r\n" + "    \"brandName\": {\r\n" + "        \"type\": \"Property\",\r\n"
+				+ "        \"value\": \"WEX\"\r\n" + "    }\r\n" + "}";
+
+		updatePayloadBadRequest = "{\r\n" + "    \"brandName\": {\r\n" + "        \"type\": \"Property\",\r\n"
+				+ "        \"value\": \"c\"\r\n" + "    },\r\n" + "    \"@context\": [\r\n"
+				+ "        \"https://uri.etsi.org/ngsi-ld/v1/ngsi-ld-core-context-v1.3.jsonld\"\r\n" + "    ]\r\n"
+				+ "}";
+
+	}
+
+	@AfterEach
+	public void teardown() {
+		payload = null;
+		updatePayload = null;
+	}
+
+	@Test
+	@Order(1)
+	public void registerCSourceTest() throws Exception {
+		ExtractableResponse<Response> response = given().body(payload)
+				.header(HttpHeaders.CONTENT_TYPE, AppConstants.NGB_APPLICATION_JSON)
+				.header(HttpHeaders.ACCEPT, AppConstants.NGB_APPLICATION_JSONLD).when()
+				.post("/ngsi-ld/v1/csourceRegistrations/").then().statusCode(Status.CREATED.getStatusCode())
+				.statusCode(201).extract();
+		int statusCode = response.statusCode();
+		assertEquals(201, statusCode);
+	}
+
+	@Test
+	@Order(2)
+	public void registerCSourceAlreadyExistTest() {
+		try {
+			ExtractableResponse<Response> response = given().body(payload)
+					.header(HttpHeaders.CONTENT_TYPE, AppConstants.NGB_APPLICATION_JSON)
+					.header(HttpHeaders.ACCEPT, AppConstants.NGB_APPLICATION_JSONLD).when()
+					.post("/ngsi-ld/v1/csourceRegistrations/").then().statusCode(Status.CONFLICT.getStatusCode())
+					.statusCode(409).extract();
+			int statusCode = response.statusCode();
+			assertEquals(409, statusCode);
+		} catch (Exception e) {
+			e.printStackTrace();
+		}
+	}
+
+	@Test
+	@Order(3)
+	public void registerCSourceBadReqestTest() {
+		try {
+			ExtractableResponse<Response> response = given().body(payload1)
+					.header(HttpHeaders.CONTENT_TYPE, AppConstants.NGB_APPLICATION_JSON)
+					.header(HttpHeaders.ACCEPT, AppConstants.NGB_APPLICATION_JSONLD).when()
+					.post("/ngsi-ld/v1/csourceRegistrations/").then().statusCode(Status.BAD_REQUEST.getStatusCode())
+					.statusCode(400).extract();
+			int statusCode = response.statusCode();
```

Comment:
No need to declare this variable so you can compare  response code directly.

Reply:
done